packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_widget_configuration.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import 'types.dart';
+
+/// A container object for configuration options when building a widget.
+///
+/// This is intended for use as a parameter in platform interface methods, to
+/// allow adding new configuration options to existing methods.
+@immutable
+class MapWidgetConfiguration {
+  /// Creates a new configuration with all the given settings.
+  const MapWidgetConfiguration({
+    required this.initialCameraPosition,
+    required this.textDirection,
```

Comment:
`maybeOf` doesn't guarantee a value though, so we'd need a hard-coded fallback-fallback. I think having that kind of logic in the app-facing package (so it can decide how to handle it at the plugin API level) makes the most sense.

Reply:
`maybeOf` doesn't guarantee a value though, so we'd need a hard-coded fallback-fallback. I think having that kind of logic in the app-facing package (so it can decide how to handle it at the plugin API level) makes the most sense.

Reply:
Ah yes, it also needs the `context` which is going to be more readily available closer to the Widget.